rollup.config.js

```diff
@@ -9,121 +9,168 @@ import copy from 'rollup-plugin-copy';
+  }),
+  json({
+    // All JSON files will be parsed by default,
+    // but you can also specifically include/exclude files
+    exclude: ['node_modules'],
+    // for tree-shaking, properties will be declared as
+    // variables, using either `var` or `const`
+    preferConst: true, // Default: false
+    // specify indentation for the generated default export —
+    // defaults to '\t'
+    indent: '  ',
+    // ignores indent and generates the smallest code
+    compact: true, // Default: false
+    // generate a named export for every property of the JSON object
+    namedExports: true, // Default: true
+  }),
+];
 
-      // specify indentation for the generated default export —
-      // defaults to '\t'
-      indent: '  ',
+export default [
+  // CommonJS & ESM
+  {
+    input: 'src/index.js',
+    preserveModules: true,
+    output: [
+      {
+        dir: 'lib',
+        name: 'CarbonAddonsIoTReact',
+        format: 'cjs',
+      },
+      {
+        dir: 'es',
+        format: 'esm',
+      },
+    ],
+    external,
+    plugins: [...plugins, ...prodSettings],
+  },
+  // Compile styles
+  {
+    input: 'src/styles.scss',
+    output: [
+      {
+        dir: 'tmp',
+        format: 'esm',
+      },
+    ],
+    plugins: [
+      postcss({
+        extract: 'lib/css/carbon-addons-iot-react.css',
+        sourceMap: true,
+        use: ['sass'],
+        plugins: [autoprefixer],
+      }),
+    ],
+  },
+  // UMD
+  {
+    input: 'src/index.js',
+    output: [
+      {
+        file: 'umd/carbon-addons-iot-react.js',
+        name: 'CarbonAddonsIoTReact',
+        format: 'umd',
+        globals: {
+          ...globals,
+        },
+      },
+    ],
+    external,
+    plugins: [
+      copy({
+        flatten: false,
+        targets: [
+          // Sass entrypoint
+          { src: 'src/styles.scss', dest: ['lib/scss', 'scss'] },
 
-      // ignores indent and generates the smallest code
-      compact: true, // Default: false
+          // Sass globals
+          {
+            src: 'src/globals',
+            dest: ['lib/scss', 'scss'],
+          },
 
-      // generate a named export for every property of the JSON object
-      namedExports: true, // Default: true
-    }),
-    ...prodSettings,
-  ],
-};
+          // Sass components
+          {
+            src: ['src/components/**/*.scss', '!src/components/**/*.story.scss'],
+            dest: ['lib/scss', 'scss'],
+          },
+          // react-resizable
+          {
+            src: [
+              'node_modules/react-resizable/css/**/*.css',
+              'node_modules/react-grid-layout/css/**/*.css',
+            ],
+            dest: ['es/node_modules', 'lib/node_modules'],
+          },
+          // Copy CSS
+          {
+            src: ['lib/css/'],
+            dest: ['./', 'umd'],
+          },
+        ],
+        verbose: env !== 'development', // logs the file copy list on production builds for easier debugging
+      }),
```

Comment:
Does this need to be part of the UMD build or could it be moved up into the `Compile styles` build section after the compilation?

Reply:
It needs to be separate from styles build unfortunately. Even though these plugins are supposed to run sequentially we are unable to copy the compiled file that is created from the postcss plugin. I put it in the umd build as to not have to have an extra build process.

Reply:
Bummer on that - makes sense though. Something to revisit for a future major version bump.